interpreter/cling/test/DynamicLibraryManager/cached_realpath.C

```diff
@@ -154,12 +154,12 @@ test_realpath("dir2/dir21/rlinkdir2"); // CHECK: OK
 test_realpath("dir2/linkdir21a1"); // CHECK: OK
 test_realpath("dir2/rlinkdir21a"); // CHECK: OK
 
-test_realpath("dir2/dir21/dir211/dir211"); // CHECK: ERROR
-test_realpath("dir2/dir21/dir211/dir211/e.txt"); // CHECK: ERROR
-test_realpath("dir2/dir21/dir211/dir211/dir211"); // CHECK: ERROR
-test_realpath("dir2/dir21/dir211/dir211/dir211/e.txt"); // CHECK: ERROR
-test_realpath("dir2/dir21/dir211/dir211/dir211/dir211"); // CHECK: ERROR
-test_realpath("dir2/dir21/dir211/dir211/dir211/dir211/e.txt"); // CHECK: ERROR
+test_realpath("dir2/dir21/dir211/dir211"); // CHECK: OK
+test_realpath("dir2/dir21/dir211/dir211/e.txt"); // CHECK: OK
+test_realpath("dir2/dir21/dir211/dir211/dir211"); // CHECK: OK
+test_realpath("dir2/dir21/dir211/dir211/dir211/e.txt"); // CHECK: OK
+test_realpath("dir2/dir21/dir211/dir211/dir211/dir211"); // CHECK: OK
+test_realpath("dir2/dir21/dir211/dir211/dir211/dir211/e.txt"); // CHECK: OK
```

Comment:
From @alexander-penev: we need `// RUN: %rm -fr %t-dir` before `// RUN: %mkdir %t-dir`.

Reply:
Yep.

Reply:
That's failing spectacularly:
```
ERROR
OK
OK
OK
realpath: ..
  err_s=0, rp_s=/home/axel/build/cling/cling-all-in-one/obj/tools/cling/test/DynamicLibraryManager/Output
  err_c=0, rp_c=/home/axel/build/cling/cling-all-in-one/obj/tools/cling/test/DynamicLibraryManager/Output/..

realpath: ../
  err_s=0, rp_s=/home/axel/build/cling/cling-all-in-one/obj/tools/cling/test/DynamicLibraryManager/Output
  err_c=0, rp_c=/home/axel/build/cling/cling-all-in-one/obj/tools/cling/test/DynamicLibraryManager/Output/..

realpath: ../.
  err_s=0, rp_s=/home/axel/build/cling/cling-all-in-one/obj/tools/cling/test/DynamicLibraryManager/Output
  err_c=0, rp_c=/home/axel/build/cling/cling-all-in-one/obj/tools/cling/test/DynamicLibraryManager/Output/..

OK
realpath: /..
  err_s=0, rp_s=/
  err_c=0, rp_c=/..

realpath: /../
  err_s=0, rp_s=/
  err_c=0, rp_c=/..

realpath: /../..
  err_s=0, rp_s=/
  err_c=0, rp_c=/..

OK
OK
OK
ERROR
ERROR
OK
OK
OK
OK
ERROR
ERROR
ERROR
OK
OK
OK
OK
OK
OK
ERROR
ERROR
ERROR
ERROR
OK
OK
realpath: dir2/linkdir211/..
  err_s=0, rp_s=/home/axel/build/cling/cling-all-in-one/obj/tools/cling/test/DynamicLibraryManager/Output/cached_realpath.C.tmp-dir/dir2/dir21
  err_c=0, rp_c=/home/axel/build/cling/cling-all-in-one/obj/tools/cling/test/DynamicLibraryManager/Output/cached_realpath.C.tmp-dir/dir2/dir21/..

realpath: dir2/rlinkdir211/..
  err_s=0, rp_s=/home/axel/build/cling/cling-all-in-one/obj/tools/cling/test/DynamicLibraryManager/Output/cached_realpath.C.tmp-dir/dir2/dir21
  err_c=0, rp_c=/home/axel/build/cling/cling-all-in-one/obj/tools/cling/test/DynamicLibraryManager/Output/cached_realpath.C.tmp-dir/dir2/dir21/..

OK
realpath: dir2/dir21/rlinkdir2
  err_s=0, rp_s=/home/axel/build/cling/cling-all-in-one/obj/tools/cling/test/DynamicLibraryManager/Output/cached_realpath.C.tmp-dir/dir2
  err_c=0, rp_c=/home/axel/build/cling/cling-all-in-one/obj/tools/cling/test/DynamicLibraryManager/Output/cached_realpath.C.tmp-dir/dir2/..

OK
OK
ERROR
ERROR
ERROR
ERROR
ERROR
ERROR
OK
OK
realpath: dir2/./backtoa.txt
  err_s=0, rp_s=/home/axel/build/cling/cling-all-in-one/obj/tools/cling/test/DynamicLibraryManager/Output/cached_realpath.C.tmp-dir/dir1/a.txt
  err_c=2, rp_c=

realpath: dir2/dir21/../backtoa.txt
  err_s=0, rp_s=/home/axel/build/cling/cling-all-in-one/obj/tools/cling/test/DynamicLibraryManager/Output/cached_realpath.C.tmp-dir/dir1/a.txt
  err_c=2, rp_c=

realpath: dir2//backtoa.txt
  err_s=0, rp_s=/home/axel/build/cling/cling-all-in-one/obj/tools/cling/test/DynamicLibraryManager/Output/cached_realpath.C.tmp-dir/dir1/a.txt
  err_c=2, rp_c=

ERROR
ERROR
ERROR
ERROR
ERROR
```
Let me push what I have so @alexander-penev and you can confirm.

Reply:
Isn't that due to #9911, works for me...